include/libtorrent/alert_types.hpp

```diff
@@ -1232,8 +1245,6 @@ namespace libtorrent
 	// listen on it.
 	struct TORRENT_EXPORT listen_failed_alert final : alert
 	{
-		enum socket_type_t { tcp, tcp_ssl, udp, i2p, socks5, utp_ssl };
```

Comment:
you should probably keep the name of this enum too. you can disambiguate by fully qualify them. ``::libtorrent::socket_t`` vs. ``::libtorrent::listen_failed_alert::socket_t``